lib/util/update.rb

```diff
@@ -118,74 +159,50 @@ def self.snapshot
-          respond 'While we will do everything we can to ensure you have a smooth experience, '
-          respond 'it is a test, and untoward things can result.  Please confirm your choice:'
-          respond 'Please confirm your participation:  ;send Y or ;send N'
-          # we are only going to get the next client-input line, and if it does not confirm, we bail
-          # we are doing this to prevent hanging the client with various other inputs by the user
-          sync_thread = $_CLIENT_ || $_DETACHABLE_CLIENT_
-          line = sync_thread.gets until line.strip =~ /^(?:<c>)?(?:;send|;s) /i
-          if line =~ /send Y|s Y/i
-            @beta_response = 'accepted'
-            respond 'Beta test installation accepted.  Thank you for considering!'
-          else
-            @beta_response = 'rejected'
-            respond 'Aborting beta test installation request.  Thank you for considering!'
-            respond
-          end
-          if @beta_response =~ /accepted/
-            filename = "https://api.github.com/repos/elanthia-online/lich-5/releases"
-            update_info = URI.parse(filename).open.read
-            record = JSON::parse(update_info).first # assumption: Latest beta release always first record in API
-            record.each { |entry, value|
-              if entry.include? 'tag_name'
-                @update_to = value.sub('v', '')
-              elsif entry.include? 'assets'
-                @holder = value
-              elsif entry.include? 'body'
-                @new_features = value.gsub(/\#\# What's Changed.+$/m, '')
-              end
-            }
-            beta_asset = @holder.find { |x| x['name'] =~ /lich-5.tar.gz/ }
-            @zipfile = beta_asset.fetch('browser_download_url')
-            Lich::Util::Update.download_update
-          elsif @beta_response =~ /rejected/
-            nil
+      def self.prep_request(version = 'latest') # default to simple update
+        case version
+        when 'latest'
+          filename = "https://api.github.com/repos/elanthia-online/lich-5/releases/latest"
+          update_info = URI.parse(filename).open.read
+          record = JSON::parse(update_info) # latest always contains one record
+        when 'beta'
+          filename = "https://api.github.com/repos/elanthia-online/lich-5/releases"
+          update_info = URI.parse(filename).open.read
+          record = JSON::parse(update_info).first # assumption: Latest beta release always first record in API
+        when VERSION_REGEX
+          requested_version = version.gsub(/v/, '') # if present, remove 'v' to standardize on version numbers
+          filename = "https://api.github.com/repos/elanthia-online/lich-5/releases"
+          update_info = URI.parse(filename).open.read
+          temp_record = JSON::parse(update_info).select { |h1| h1['tag_name'] == "v#{requested_version}" }
+          unless temp_record.nil? or temp_record.empty?
+            record = temp_record.first
```

Comment:
Which is what is processed in the next three lines.  If the requested version is not in the releases api, it fails out, with an error message.

Reply:
This goes to the first review comment - I mistyped something, I am sure.